wamr-compiler/CMakeLists.txt

```diff
@@ -76,7 +78,9 @@ add_definitions(${LLVM_DEFINITIONS})
 message(STATUS "Found LLVM ${LLVM_PACKAGE_VERSION}")
 message(STATUS "Using LLVMConfig.cmake in: ${LLVM_DIR}")
 
-set (CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -Wl,--gc-sections")
+if (NOT WAMR_BUILD_PLATFORM MATCHES "darwin")
```

Comment:
Actually it is not caused by darwin platform, it is caused by clang compiler, so we suggest modifying it as:

if (NOT (CMAKE_C_COMPILER MATCHES ".*clang.*" OR CMAKE_C_COMPILER_ID MATCHES ".*Clang"))
  set (CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -Wl,--gc-sections")
endif()

Similar to product-mini/platforms/linux/CMakeLists.txt